src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +23,141 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
-*/
+void BgLotus_Init(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+    s32 pad;
+    s32 sp2C;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gLilyPadCol);
+    this->dyna.actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->dyna.actor.floorPoly, &sp2C,
+                                                               &this->dyna.actor, &this->dyna.actor.world.pos);
+    this->unk168 = 0x60;
+    this->dyna.actor.world.rot.y = (s32)Rand_Next() >> 0x10;
+    this->actionFunc = func_80AD68DC;
+}
+
+void BgLotus_Destroy(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AD6830(BgLotus* this) {
+    f32 temp_fv1;
+
+    if (this->dyna.actor.params == 0) {
+        temp_fv1 = Math_SinF(this->unk166 * (M_PI / 4)) * ((0.014f * (this->unk166 / (80.0f * 0.1f))) + 0.01f);
+        this->dyna.actor.scale.x = ((1.0f + temp_fv1) * 0.1f);
+        this->dyna.actor.scale.z = ((1.0f - temp_fv1) * 0.1f);
+    }
+}
+
+void func_80AD68DC(BgLotus* this, PlayState* play) {
+    f32 sp34;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+    this->unk168--;
+    sp34 = Math_SinF(this->unk168 * 0.06544985f) * 6.0f;
```

Comment:
```suggestion
    sp34 = Math_SinF(this->unk168 * (M_PI / 48)) * 6.0f;
```
As mentioned in the comment